src/test/java/org/scijava/io/handle/FileHandleTest.java

```diff
@@ -108,4 +110,30 @@ public void testNotCreatedByClose() throws IOException {
 		handle.close();
 		assertFalse(nonExistentFile.exists());
 	}
+
+	@Test
+	public void testNotCreatedByRead() throws IOException {
+		final Context ctx = new Context();
+		final DataHandleService dhs = ctx.service(DataHandleService.class);
+
+		final File nonExistentFile = //
+			File.createTempFile("FileHandleTest", "none xistent file");
+		final URI uri = nonExistentFile.toURI();
+		assertTrue(nonExistentFile.delete());
+		assertFalse(nonExistentFile.exists());
+
+		final FileLocation loc = new FileLocation(uri);
+		final DataHandle<?> handle = dhs.create(loc);
+		assertFalse(handle.exists());
+		try {
+			handle.read(); // this will fail as there is no underlying file!
+			fail("Read successfully from non-existing file!");
+		}
+		catch (final Exception exc) {
```

Comment:
How about catching only `IOException` instead of `Exception` here?

Reply:
Good point, I just pushed an updated version that does just that.